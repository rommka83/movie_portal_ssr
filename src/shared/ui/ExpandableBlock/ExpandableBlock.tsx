import React, { useState } from 'react';
import styles from './expandableblock.module.css';
import { useTranslation } from 'i18n';
import classNames from 'classnames';

interface IExpandableBlock {
  text: string;
}

export const ExpandableBlock = ({ text }: IExpandableBlock) => {
  const { t } = useTranslation();
  const [expanded, setExpanded] = useState(false);
  const onExpand = () => {
    setExpanded((prev) => !prev);
  };
  const buttonText = expanded ? t('CatalogPageHeader.Collapse') : t('CatalogPageHeader.Expand');
  return (
    <div className={styles.container}>
      <p
        className={classNames(styles.text, {
          [styles.expanded]: expanded,
        })}
        dangerouslySetInnerHTML={{ __html: text }}
      />

      <button className={styles.expandButton} onClick={onExpand}>
        {buttonText}
      </button>
    </div>
  );
};
