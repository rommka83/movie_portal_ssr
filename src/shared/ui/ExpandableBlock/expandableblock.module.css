.container {
  display: flex;
  flex-direction: column;
  gap: 8px;
  align-items: start;
  width: 75%;
}

@media (max-width: 1160px) {
  .container {
    width: 100%;
  }
}

.text:not(.expanded) {
  display: -webkit-box;
  overflow: hidden;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  text-overflow: ellipsis;
}

.text {
  display: block;
  color: var(--gray-light);
  font-size: 15px;
  line-height: 20px;
}

.expandButton {
  color: var(--white);
  font-size: 15px;
  line-height: 20px;
  font-weight: 500;
}
