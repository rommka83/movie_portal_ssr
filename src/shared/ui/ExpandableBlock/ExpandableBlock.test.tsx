import renderer from 'react-test-renderer';
import { ExpandableBlock } from './ExpandableBlock';

describe('проверка ExpandableBlock', () => {
  test('отрисовка ExpandableBlock', () => {
    const component = renderer.create(<ExpandableBlock text='Проверка' />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
