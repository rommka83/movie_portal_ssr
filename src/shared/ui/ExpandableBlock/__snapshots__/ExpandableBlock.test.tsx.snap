// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка ExpandableBlock отрисовка ExpandableBlock 1`] = `
<div
  className="container"
>
  <p
    className="text"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Проверка",
      }
    }
  />
  <button
    className="expandButton"
    onClick={[Function]}
  >
    Expand
  </button>
</div>
`;
